web-playground/e2e/sample-tester.test.ts

```diff
@@ -0,0 +1,82 @@
+import { FrameWaitForFunctionOptions, Page } from "puppeteer";
+import * as fs from 'fs';
+import * as path from 'path';
+
+describe('test all grammars with puppeteer', () => {
+    beforeEach(async () => {
+        await page.goto('http://localhost:3000/playground');
```

Comment:
Maybe this negates my first comment: The test will wait for the playground to be ready. What happens, though, if the playground is never ready (if it broke, say)? Does `beforeEach` have an implicit timeout?

Reply:
Then the test quickly fails with `net::ERR_CONNECTION_REFUSED at http://localhost:3000/playground`.